modules/ingestor/src/service/format.rs

```diff
@@ -50,34 +52,39 @@ impl<'g> Format {
             Format::CSAF => {
                 // issuer is internal as publisher of the document.
                 let loader = CsafLoader::new(graph);
-                let csaf: Csaf = from_stream(stream).await?;
+                let csaf: Csaf = json_from_stream(stream).await?;
                 loader.load(labels, csaf, digests).await
             }
             Format::OSV => {
                 // issuer is :shrug: sometimes we can tell, sometimes not :shrug:
                 let loader = OsvLoader::new(graph);
-                let osv: Vulnerability = from_stream(stream).await?;
+                let osv: Vulnerability = json_from_stream(stream).await?;
                 loader.load(labels, osv, digests, issuer).await
             }
             Format::CVE => {
                 // issuer is always CVE Project
                 let loader = CveLoader::new(graph);
-                let cve: Cve = from_stream(stream).await?;
+                let cve: Cve = json_from_stream(stream).await?;
                 loader.load(labels, cve, digests).await
             }
             Format::SPDX => {
                 let loader = SpdxLoader::new(graph);
-                let v: Value = from_stream(stream).await?;
+                let v: Value = json_from_stream(stream).await?;
                 loader.load(labels, v, digests).await
             }
             Format::CycloneDX => {
                 let loader = CyclonedxLoader::new(graph);
-                let v: Value = from_stream(stream).await?;
+                let v: Value = json_from_stream(stream).await?;
                 let sbom = Bom::parse_json_value(v)
                     .map_err(|err| Error::UnsupportedFormat(format!("Failed to parse: {err}")))?;
 
                 loader.load(labels, sbom, digests).await
             }
+            Format::ClearlyDefined => {
+                let loader = ClearlyDefinedLoader::new(graph);
+                let v: serde_yml::Value = yaml_from_stream(stream).await?;
```

Comment:
Why not just create the Curation? Why parse it once here and then again inside the loader?
```suggestion
                let v: Curation = yaml_from_stream(stream).await?;
```

Reply:
Good call.